docs/developer.md

```diff
@@ -0,0 +1,77 @@
+# DEVELOPER GUIDE --- WIP
+
+## Contributing
+// TODO(user): Add detailed information on how you would like others to contribute to this project
+
+### How it works
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
+
+
+
+### Running on the cluster
+1. Install Instances of Custom Resources:
+```sh
+kubectl apply -f config/samples/
+```
+2. Build and push your image to the location specified by `IMG`:
+```sh
+make docker-build docker-push IMG=<some-registry>/backstage-operator:tag
+```
+3. Deploy the controller to the cluster with the image specified by `IMG`:
+```sh
+make deploy IMG=<some-registry>/backstage-operator:tag
+```
+### Uninstall CRDs
+To delete the CRDs from the cluster:
+```sh
+make uninstall
+```
+### Undeploy controller
+UnDeploy the controller from the cluster:
+```sh
+make undeploy
+```
+## Build and Deploy with OLM
```

Comment:
```suggestion
### Build and Deploy with OLM
```

This should be at the same level as the rest of the topics, right? Otherwise, the table of contents seems strange to me:

![image](https://github.com/janus-idp/operator/assets/593208/2f140a33-abf7-4df3-b7d4-0ae00bcc9f58)

Reply:
I think so, it was generated in fact :)